plugin/trino-kafka/src/main/java/io/trino/plugin/kafka/KafkaInternalFieldManager.java

```diff
@@ -131,62 +155,87 @@ ColumnMetadata getColumnMetadata(boolean hidden)
-                        BigintType.BIGINT))
-                .put(OFFSET_TIMESTAMP_FIELD, new InternalField(
-                        OFFSET_TIMESTAMP_FIELD,
-                        "Message timestamp",
-                        TIMESTAMP_MILLIS))
-                .buildOrThrow();
+        String prefix = kafkaConfig.getInternalFieldPrefix();
+        ImmutableList<InternalField> fields = Stream.of(
+                        new InternalField(
+                                PARTITION_ID_FIELD,
+                                prefix + "partition_id",
+                                "Partition Id",
+                                BigintType.BIGINT),
+                        new InternalField(
+                                PARTITION_OFFSET_FIELD,
+                                prefix + "partition_offset",
+                                "Offset for the message within the partition",
+                                BigintType.BIGINT),
+                        new InternalField(
+                                MESSAGE_CORRUPT_FIELD,
+                                prefix + "message_corrupt",
+                                "Message data is corrupt",
+                                BooleanType.BOOLEAN),
+                        new InternalField(
+                                MESSAGE_FIELD,
+                                prefix + "message",
+                                "Message text",
+                                createUnboundedVarcharType()),
+                        new InternalField(
+                                HEADERS_FIELD,
+                                prefix + "headers",
+                                "Headers of the message as map",
+                                varcharMapType),
+                        new InternalField(
+                                MESSAGE_LENGTH_FIELD,
+                                prefix + "message_length",
+                                "Total number of message bytes",
+                                BigintType.BIGINT),
+                        new InternalField(
+                                KEY_CORRUPT_FIELD,
+                                prefix + "key_corrupt",
+                                "Key data is corrupt",
+                                BooleanType.BOOLEAN),
+                        new InternalField(
+                                InternalFieldId.KEY_FIELD,
+                                prefix + "key",
+                                "Key text",
+                                createUnboundedVarcharType()),
+                        new InternalField(
+                                KEY_LENGTH_FIELD,
+                                prefix + "key_length",
+                                "Total number of key bytes",
+                                BigintType.BIGINT),
+                        new InternalField(
+                                OFFSET_TIMESTAMP_FIELD,
+                                prefix + "timestamp",
```

Comment:
maybe instead of adding a prefix here, let's check if prefix is added on the getFieldByName() ?

Reply:
That would mean that `InternalField#getColumnName` returns not the actual name, but only a `root` of the name, so actually it's not usable if used alone without making prefix visible outside.

May be there is a way of general improvement of usages of `getColumnName` but I'd prefer not to.